build-tools/src/main/kotlin/org/jetbrains/kotlin/CompilationDatabase.kt

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package org.jetbrains.kotlin
+
+import java.io.File
+import javax.inject.Inject
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.builtins.list
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonConfiguration
+import org.gradle.api.GradleException
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import org.jetbrains.kotlin.konan.target.HostManager
+
+@Serializable
```

Comment:
IIRC, we had strong opinion against using `kotlinx.serialization` in our build scripts.
@PavelPunegov probably knows more.

Reply:
Well, we still use them, but I think it's better to avoid to do that. It's ok for now, as I'm going to rewrite that and have started actually.

Reply:
What's the preferred way to deal JSON then?

Reply:
There is a https://github.com/JetBrains/kotlin-native/tree/master/tools/benchmarks/shared/src/main/kotlin/org/jetbrains/report/json
but it requires to be moved into `build-tools`. And that I was doing but haven't finished yet.